grails-app/views/stockMovement/show.gsp

```diff
@@ -212,12 +212,11 @@
                                     <g:message code="order.label"/>
                                 </td>
                                 <td class="value">
-                                    <g:each var="order" in="${stockMovement?.shipment?.shipmentItems*.orderItems?.order?.flatten()}">
-                                        <g:link controller="order" action="show" id="${order?.id}" params="[override:true]">
-                                            ${g.message(code:'default.view.label', args: [g.message(code: 'order.label')])}
-                                            ${order.orderNumber}
-                                        </g:link>
-                                    </g:each>
+                                    <g:set var="order" value="${stockMovement?.shipment?.shipmentItems*.orderItems?.order?.flatten()?.first()}"/>
```

Comment:
What if order items will have different order? I believe, you @jmiranda said that this is possible? I think it would be better to have order?.flatten().unique() instead of .first()

Reply:
That's a good point.

Reply:
Let's move the traversal code to Shipment. So Shipment.getOrderItems(), Shipment.getOrders() Shipment.getOrderByStatus(). Like @awalkowiak did for Orders.

Reply:
@jmiranda I've added this change